pkg/controller/servicebindingrequest/binder.go

```diff
@@ -501,48 +503,51 @@ func (b *Binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 		}
 
 		log.Debug("Updating object...")
-		if err := b.client.Update(b.ctx, updatedObj); err != nil {
+		gk := updatedObj.GroupVersionKind().GroupKind()
+		version := updatedObj.GroupVersionKind().Version
+		mapping, err := b.restMapper.RESTMapping(gk, version)
+		if err != nil {
 			return nil, err
 		}
+		updated, err := b.dynClient.Resource(mapping.Resource).
+			Namespace(updatedObj.GetNamespace()).
+			Update(updatedObj, metav1.UpdateOptions{})
 
-		log.Debug("Reading back updated object...")
-		// reading object back again, to comply with possible modifications
-		namespacedName := types.NamespacedName{
-			Namespace: updatedObj.GetNamespace(),
-			Name:      updatedObj.GetName(),
-		}
-		if err = b.client.Get(b.ctx, namespacedName, updatedObj); err != nil {
+		if err != nil {
 			return nil, err
 		}
-
-		updatedObjs = append(updatedObjs, updatedObj)
+		updatedObjs = append(updatedObjs, updated)
 	}
 
 	return updatedObjs, nil
 }
 
 // remove attempts to update each given object without any service binding related information.
-func (b *Binder) remove(objs *unstructured.UnstructuredList) error {
-	for _, obj := range objs.Items {
+func (b *Binder) remove(objs []*unstructured.Unstructured) error {
```

Comment:
Is this change from `*unstructured.UnstructuredList` to `[]*unstructured.Unstructured` essential for this change?